src/main/java/kostka/moviecatalog/service/DbMovieService.java

```diff
@@ -8,19 +8,24 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.EnableAsync;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
 import java.util.stream.Collectors;
 
 import static kostka.moviecatalog.service.rabbitmq.RabbitMqReceiver.ALL_MOVIES_KEY;
 import static kostka.moviecatalog.service.rabbitmq.RabbitMqReceiver.LATEST_MOVIES_KEY;
 import static kostka.moviecatalog.service.rabbitmq.RabbitMqReceiver.TOP_RATING_KEY;
 
 @Service
+@EnableAsync
```

Comment:
EnableAsync podle mě funguje pro celou aplikaci. Takže stačí jen jednou.
Ale možná se pletu.

Podle nějaké dokumentace se používá spolu s @Configuration anotací a zapíná zpracování @Async anotací.
Takže by to u MovieCatalogApplication mělo stačit.

Reply:
jo mas pravdu, jedno povoleni Async staci pro cely projekt, diky (y)